src/core/core.layoutService.js

```diff
@@ -10,24 +10,58 @@ module.exports = function(Chart) {
 	Chart.layoutService = {
 		defaults: {},
 
-		// Register a box to a chartInstance. A box is simply a reference to an object that requires layout. eg. Scales, Legend, Plugins.
+		/**
+		 * Register a box to a chartInstance.
+		 * A box is simply a reference to an object that requires layout. eg. Scales, Legend, Title.
+		 * @method Chart.layoutService.addBox
+		 * @param {Core.Controller} chartInstance the chart to use
+		 * @param {Object} box the box to add
+		 * @param {Number} box.order the order of the boxes. Higher numbers are further away from the chart area
+		 * @param box.options Options for configuring the box
+		 * @param {String} box.options.position the position of the box. Can be 'left', 'top', 'right', 'bottom', or 'chartArea'
+		 * @param {Boolean} box.options.fullWidth if true, and the box is horizontal then push vertical boxes down
+		 * @param {Function} box.isHorizontal return true if this is a horizontal box (top or bottom position)
+		 * @param {Function} box.update Takes two parameters: width and height. Returns size of box
+		 * @param {Function} box.getPadding Returns an object with padding on the edges
+		 * @param {Number} box.width Width of box. Must be valid after update()
+		 * @param {Number} box.height Height of box. Must be valid after update()
+		 * @param {Number} box.left Left edge of the box. Set by layout system and cannot be used in update
+		 * @param {Number} box.top Top edge of the box. Set by layout system and cannot be used in update
+		 * @param {Number} box.right Right edge of the box. Set by layout system and cannot be used in update
+		 * @param {Number} box.bottom Bottom edge of the box. Set by layout system and cannot be used in update
+		 */
 		addBox: function(chartInstance, box) {
 			if (!chartInstance.boxes) {
 				chartInstance.boxes = [];
 			}
+
+			// Ensure that all boxes have an order
+			if (!box.order) {
+				box.order = 0;
```

Comment:
Default `order` should be 0, because boxes can be registered from anywhere, so it would be impossible to really control the box order:

```javascript
addBox(chart, {});              // foo (index 0)
addBox(chart, {order: 1});      // bar (index 1)
addBox(chart, {});              // xyz (index 2)
```

Effective order is:
- case default to 0: `[foo, xyz, bar]` (**expected**)
- case auto-increment: `[foo, bar, xyz]`

Instead, we could use the box index as second  sort criteria to obtain a stable sort:
```javascript
function sort(array, reverse) {
    return array.map(function(v, i) {
        v._tmp_index_ = i;
    })
    .sort(function (a, b) {
        var v0 = reverse? b : a;
        var v1 = reverse? a : b;
        return v0.order === v1.order?
            v0._tmp_index_ - v1._tmp_index_ :
            v0.order - v1.order;
    })
    .map(function(v) {
        delete v._tmp_index_;
    });
}

update: function() {
    sort(leftBoxes, false);
    sort(rightBoxes, true);
    sort(topBoxes, false);
    sort(bottomBoxes, true);
}
```

Not super efficient, but I don't think there will be so much boxes to sort!